modules/auxiliary/scanner/smtp/smtp_relay.rb

```diff
@@ -84,6 +84,11 @@ def do_test_relay(testnumber, mailfrom, mailto)
 
       res = raw_send_recv("EHLO X\r\n")
       vprint_status("#{res.inspect}")
+      # check if the EHLO is actually supported. In case it's not, try the HELO command instead
+      if res.inspect&.include?('Command EHLO not known')
```

Comment:
We should probably be checking status code here, not the human readable text

https://www.rfc-editor.org/rfc/rfc5321.html#section-4.2.1

```
There are four values for the first digit of the reply code:

   2yz  Positive Completion reply
      The requested action has been successfully completed.  A new
      request may be initiated.

   3yz  Positive Intermediate reply
      The command has been accepted, but the requested action is being
      held in abeyance, pending receipt of further information.  The
      SMTP client should send another command specifying this
      information.  This reply is used in command sequence groups (i.e.,
      in DATA).

   4yz  Transient Negative Completion reply
      The command was not accepted, and the requested action did not
      occur.  However, the error condition is temporary, and the action
      may be requested again.  The sender should return to the beginning
      of the command sequence (if any).  It is difficult to assign a
      meaning to "transient" when two different sites (receiver- and
      sender-SMTP agents) must agree on the interpretation.  Each reply
      in this category might have a different time value, but the SMTP
      client SHOULD try again.  A rule of thumb to determine whether a
      reply fits into the 4yz or the 5yz category (see below) is that
      replies are 4yz if they can be successful if repeated without any
      change in command form or in properties of the sender or receiver
      (that is, the command is repeated identically and the receiver
      does not put up a new implementation).

   5yz  Permanent Negative Completion reply
      The command was not accepted and the requested action did not
      occur.  The SMTP client SHOULD NOT repeat the exact request (in
      the same sequence).  Even some "permanent" error conditions can be
      corrected, so the human user may want to direct the SMTP client to
      reinitiate the command sequence by direct action at some point in
      the future (e.g., after the spelling has been changed, or the user
      has altered the account status).
```

So 5\d\d at the start is probably enough for a downgrade to HELO, or at the very least a non-2\d\d

Reply:
@adfoster-r7 yeah I agree, response codes would be better. I have pushed a change to check if the server response starts with `"\5\d\d` because all the responses I get back are wrapped in double quotes. I'm not 100% if this will always be the case and I have no other target to test this on atm, so it would be good to double check this.